editor/plugins/gdextension/gdextension_edit_dialog.cpp

```diff
@@ -0,0 +1,244 @@
+		}
+	}
+}
+
+void GDExtensionEditDialog::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY: {
+			connect(SceneStringName(confirmed), callable_mp(this, &GDExtensionEditDialog::_on_confirmed));
+			get_cancel_button()->connect(SceneStringName(pressed), callable_mp(this, &GDExtensionEditDialog::_on_canceled));
+		} break;
+	}
+}
+
+GDExtensionEditDialog::GDExtensionEditDialog() {
+	get_ok_button()->set_disabled(true);
+	get_ok_button()->set_text(TTR("Save"));
+	set_hide_on_ok(true);
+	set_title(TTR("Edit GDExtension"));
+	set_min_size(Size2(400, 300) * EDSCALE);
+
+	VBoxContainer *vbox = memnew(VBoxContainer);
+	vbox->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	vbox->set_v_size_flags(Control::SIZE_EXPAND_FILL);
+	add_child(vbox);
+
+	ScrollContainer *scroll = memnew(ScrollContainer);
+	scroll->set_custom_minimum_size(Size2(400, 150) * EDSCALE);
+	scroll->set_horizontal_scroll_mode(ScrollContainer::SCROLL_MODE_DISABLED);
+	scroll->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	scroll->set_v_size_flags(Control::SIZE_EXPAND_FILL);
+	vbox->add_child(scroll);
+
+	GridContainer *grid = memnew(GridContainer);
+	grid->set_columns(2);
+	grid->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->set_v_size_flags(Control::SIZE_SHRINK_BEGIN);
+	scroll->add_child(grid);
+
+	// GDExtension file path (read-only).
+	Label *gdextension_path_label = memnew(Label);
+	gdextension_path_label->set_text(TTR("Path:"));
+	gdextension_path_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(gdextension_path_label);
+
+	gdextension_path = memnew(Label);
+	gdextension_path->set_text("res://addons/my_extension/my_extension.gdextension");
+	gdextension_path->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(gdextension_path);
+
+	// Entry symbol.
+	Label *entry_symbol_label = memnew(Label);
+	entry_symbol_label->set_text(TTR("Entry Symbol:"));
+	entry_symbol_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(entry_symbol_label);
+
+	entry_symbol_edit = memnew(LineEdit);
+	entry_symbol_edit->set_placeholder("libname_library_init");
+	entry_symbol_edit->set_tooltip_text(TTR("Required. The symbol to use as the entry point for the extension."));
+	entry_symbol_edit->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(entry_symbol_edit);
+
+	// Compatibility Min Version
+	Label *compat_min_version_label = memnew(Label);
+	compat_min_version_label->set_text(TTR("Compat Min Version:"));
+	compat_min_version_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(compat_min_version_label);
+
+	compat_min_version_edit = memnew(LineEdit);
+	compat_min_version_edit->set_placeholder(VERSION_BRANCH);
+	compat_min_version_edit->set_tooltip_text(TTR("Required. The minimum version of Godot that the extension is compatible with."));
```

Comment:
I'm not 100% sure it makes sense to allow the user to choose this freely. People may try to enter `4.1` and then be confused that it doesn't work, given they've used the godot-cpp branch `4.4`. Same for maximum.

For now, I would recommend setting min to the current godot version (and not letting the user choose), and max to 5.0.
In a future version, you could have a dropdown that lets the user choose a godot-cpp version (e.g. `4.3`), and set the min version automatically accordingly. I think this may be too much trouble for the initial version of this PR though.

(If https://github.com/godotengine/godot-proposals/issues/10243 is carried through, this will probably be easier in the future)

Reply:
Again this is used to edit existing GDExtensions.